### PR TITLE
Moves __init__ docstrings to class docstrings and removes __init__ from docs, add missing docstrings, and general cleanup

### LINKED ISSUE #2392: Add `__init__` docstrings to main class page
Currently on the main page for each class we can only see the `__init__` [parameters but not their respective docstrings.](https://evalml.alteryx.com/en/stable/generated/evalml.automl.AutoMLSearch.html#evalml.automl.AutoMLSearch) To do, one needs to click on the `__init__` method [itself](https://evalml.alteryx.com/en/stable/generated/methods/evalml.automl.AutoMLSearch.__init__.html#evalml.automl.AutoMLSearch.__init__). This can be improved by adding it to the main page like [sklearn does.](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ShuffleSplit.html)

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #2392

This PR does the following:
- Moves `__init__` docstrings to class docstrings and removes `__init__` from docs.
- Adds docstrings that were missing (mostly estimators + some other classes)
- Updates docstrings to include default values if missing.
- Deletes some __init__ methods that were not necessary: some init methods only had a docstring attached or called super()'s; I suspect this had to do with wanting a docstring for the class. Since I'm moving them to the class docstring, we can now delete these! (see data checks + ensemble estimators)

Example of how this change looks on a specific class: https://feature-labs-inc-evalml--2452.com.readthedocs.build/en/2452/generated/evalml.pipelines.components.StackedEnsembleClassifier.html#evalml.pipelines.components.StackedEnsembleClassifier

vs main: https://evalml.alteryx.com/en/stable/generated/evalml.pipelines.components.StackedEnsembleClassifier.html

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -266,6 +265,13 @@ class AccessorMethodDocumenter(AccessorLevelDocumenter, MethodDocumenter):
     # lower than MethodDocumenter so this is not chosen for normal methods
     priority = 0.6
 
+def autodoc_skip_member(app, what, name, obj, skip, options):
```
Comment (angela97lin) [inline docs/source/conf.py:268]: Hide private methods and attributes

File: evalml/automl/engine/sequential_engine.py [hunk 2]
```diff
@@ -19,6 +19,11 @@ class SequentialComputation(EngineComputation):
     """
 
     def __init__(self, work, **kwargs):
+        """An Engine that sequentially computes the submitted jobs.
+
+        Arguments:
```
Comment (freddyaboulton) [inline evalml/automl/engine/sequential_engine.py:24]: Shouldn't we move the args section to the class docstring?

File: evalml/data_checks/invalid_targets_data_check.py [hunk 3]
```diff
@@ -35,7 +43,7 @@ def __init__(self, problem_type, objective, n_unique=100):
                 e.g. 'binary', 'multiclass', 'regression, 'time series regression'
             objective (str or ObjectiveBase): Name or instance of the objective class.
             n_unique (int): Number of unique target values to store when problem type is binary and target
-                incorrectly has more than 2 unique values. Non-negative integer. Defaults to 100. If None, stores all unique values.
+                incorrectly has more than 2 unique values. Non-negative integer. If None, stores all unique values. Defaults to 100.
```
Comment (freddyaboulton) [inline evalml/data_checks/invalid_targets_data_check.py:46]: I think we can delete this init docstring right?

File: evalml/data_checks/no_variance_data_check.py [hunk 4]
```diff
@@ -13,7 +13,13 @@
...
+    """Check if the target or any of the features have no variance.
+
+    Arguments:
+        count_nan_as_value (bool): If True, missing values will be counted as their own unique value.
+            If set to True, a feature that has one unique value and all other data is missing, a
+            DataCheckWarning will be returned instead of an error. Defaults to False.
+    """
 
     def __init__(self, count_nan_as_value=False):
         """Check if the target or any of the features have no variance.
```
Comment (freddyaboulton) [inline evalml/data_checks/no_variance_data_check.py:25]: I think we can delete this init doctring?

File: evalml/tuners/skopt_tuner.py [hunk 5]
```diff
@@ -12,7 +12,12 @@
...
-    """Bayesian Optimizer."""
+    """Bayesian Optimizer.
+
+    Arguments:
+        pipeline_hyperparameter_ranges (dict): A set of hyperparameter ranges corresponding to a pipeline's parameters
+        random_seed (int): The seed for the random number generator. Defaults to 0.
+    """
 
     def __init__(self, pipeline_hyperparameter_ranges, random_seed=0):
         """Init SkOptTuner
```
Comment (freddyaboulton) [inline evalml/tuners/skopt_tuner.py:23]: I think we can delete this docstring

**Review by freddyaboulton — APPROVED** [review]
@angela97lin Thanks for adding all these missing docstrings! I think there are some init docstrings we can delete now but apart from that this LGTM!

File: evalml/data_checks/no_variance_data_check.py [hunk 6]
```diff
@@ -13,7 +13,13 @@
 
 
 class NoVarianceDataCheck(DataCheck):
-    """Check if the target or any of the features have no variance."""
+    """Check if the target or any of the features have no variance.
+
+    Arguments:
+        count_nan_as_value (bool): If True, missing values will be counted as their own unique value.
+            If set to True, a feature that has one unique value and all other data is missing, a
```
Comment (bchen1116) [inline evalml/data_checks/no_variance_data_check.py:20]: This second sentence is a little confusing. 
`If set to True, a feature that has one unique value and all other data is missing, a DataCheckWarning will be returned instead of an error.` -> `Additionally, if true, will return a DataCheckWarning instead of an error if the feature has mostly missing data and only one unique value.`?

File: evalml/pipelines/components/transformers/column_selectors.py [hunk 7]
```diff
@@ -5,12 +5,15 @@
 
 
 class ColumnSelector(Transformer):
-    def __init__(self, columns=None, random_seed=0, **kwargs):
-        """Initalizes an transformer that drops specified columns in input data.
+    """
```
Comment (bchen1116) [inline evalml/pipelines/components/transformers/column_selectors.py:8]: Why do some lines have the quotes on one line and the description on the next, like the line here, while others are on the same line (like the description for `SelectColumns` on line 97)?

File: evalml/tuners/skopt_tuner.py [hunk 8]
```diff
@@ -12,7 +12,12 @@
 
 
 class SKOptTuner(Tuner):
-    """Bayesian Optimizer."""
+    """Bayesian Optimizer.
+
+    Arguments:
+        pipeline_hyperparameter_ranges (dict): A set of hyperparameter ranges corresponding to a pipeline's parameters
```
Comment (bchen1116) [inline evalml/tuners/skopt_tuner.py:18]: period?

File: evalml/tuners/tuner.py [hunk 9]
```diff
@@ -4,22 +4,21 @@
 
 
 class Tuner(ABC):
-    """Defines API for Tuners.
+    """Defines API for base Tuner classes.
 
     Tuners implement different strategies for sampling from a search space. They're used in EvalML to search the space of pipeline hyperparameters.
+
+    Arguments:
+        pipeline_hyperparameter_ranges (dict): a set of hyperparameter ranges corresponding to a pipeline's parameters
```
Comment (bchen1116) [inline evalml/tuners/tuner.py:12]: period?

**Review by bchen1116 — APPROVED** [review]
Wow, great work getting the docstrings all updated and cleaned up! This is huge, and the docs look real good. 

Left a few comments on potential things to clean, but nothing blocking!

File: evalml/pipelines/components/transformers/column_selectors.py (hunk 7 above)
Comment (angela97lin) [reply evalml/pipelines/components/transformers/column_selectors.py:8]: Ah, I don't think there's a difference tbh :d
